scaper/core.py

```diff
@@ -1913,29 +1939,86 @@ def _generate_audio(self, audio_path, ann, reverb=None,
                         'Unsupported event role: {:s}'.format(
                             e.value['role']))
 
+            # Finally combine all the files and optionally apply reverb.
+            # If there are no events, throw a warning.
+            if len(event_audio_list) == 0:
+                warnings.warn(
+                    "No events to synthesize (silent soundscape), no audio "
+                    "generated.", ScaperWarning)
+            else:                        
+                tfm = sox.Transformer()
+                if reverb is not None:
+                    tfm.reverb(reverberance=reverb * 100)
+
+                # Sum all events to get soundscape audio
+                soundscape_audio = sum(event_audio_list)
+
+                # Check for clipping and fix [optional]
+                max_sample = np.max(np.abs(soundscape_audio))
+                clipping = max_sample > 1
+                if clipping:
+                    warnings.warn('Soundscape audio is clipping!',
+                                  ScaperWarning)
+
+                if peak_normalization or (clipping and fix_clipping):
+
+                    # normalize soundscape audio and scale event audio
+                    soundscape_audio, event_audio_list, scale_factor = \
+                        peak_normalize(soundscape_audio, event_audio_list)
+
+                    warnings.warn(
+                        'Peak normalization applied (scale factor = {})'.format(
```

Comment:
Can we also report the change that happened to `ref_db`? It would be `20*np.log10(scale_factor)`.

Reply:
done